src/com/hms/flexyosisoftconnector/OSIsoftConfig.java

```diff
@@ -146,9 +185,103 @@ public static void initConfig(String configFile) throws JSONException, IOExcepti
             + "&Content-Type=application/json"
             + "&X-Requested-With=JSONHttpRequest"
             + "&messageformat=json&omfversion=1.1";
+    ocsPostHeaders =
+        "Authorization=Bearer "
+            + OSIsoftConfig.getOcsToken()
+            + "&Content-Type=application/json"
+            + "&messageformat=json&omfversion=1.1"
+            + "&Producertoken="
+            + OSIsoftConfig.getOcsToken();
+
     setFlexyName();
   }
 
+  /**
+   * The token will be retrieved in a BASIC script due to issues with the request when made in Java.
+   * Assume the response file exists and read from it.
+   */
+  private static void setOCSToken() {
+    // file name hardcoded in basic script and should never change.
+    String tokenFile = "/usr/token.json";
+    JSONTokener JsonT = null;
+    String token = "";
+    try {
+      JsonT = new JSONTokener(FileAccessManager.readFileToString(tokenFile));
+      // Build a JSON Object containing the whole file
+      JSONObject tokenJSON = new JSONObject(JsonT);
+
+      token = tokenJSON.getString("access_token");
+    } catch (IOException e) {
+      Logger.LOG_SERIOUS("Unable to read OCS token from https request response file.");
+      Logger.LOG_EXCEPTION(e);
+    } catch (JSONException e) {
+      Logger.LOG_SERIOUS(
+          "Unable to parse JSON in OCS token response file from OCS new token https request.");
+      Logger.LOG_EXCEPTION(e);
+    }
+
+    // Set the access token
+    OcsOmfToken = token;
+  }
+
+  /**
+   * This function need to have a basic script running that is specified in the README. Updates the
+   * OCS token if the OCS token tag is not set to 0. This tag will then be reset to 0.
+   */
+  public static void updateOcsToken() {
+
+    /* Ensure the JVM and basic script are working together by checking the value of a tag is 0.
+     * The basic script will increment the tag value by 1 every time a new request is sent.
+     * The Java program should reset that tag to 0 after the request's response file is handled.
+     */
+    final int TAG_INITIAL_VALUE = 0;
+
+    // check tag to see if it is {@link TAG_INITIAL_VALUE}
+    TagControl tc = null;
+    try {
+      tc = new TagControl("tokenReq");
+
+    } catch (EWException e) {
+      Logger.LOG_SERIOUS("Unable to read OCS token requester tag.");
+      Logger.LOG_SERIOUS("Please make sure the tag 'tokenReq' exists and is set to integer.");

```

Comment:
you may want to specify what about the tag needs to be set to integer as you could be talking about datatype or value here.